test/ibm/test_ibm_job_states.py

```diff
@@ -400,7 +400,8 @@ def run_with_api(self, api):
         """Creates a new ``IBMJob`` running with the provided API object."""
         backend = IBMBackend(mock.Mock(), mock.Mock(), mock.Mock(), api_client=api)
         self._current_api = api
-        self._current_qjob = backend.run(qobj=FakeQobj())  # pylint: disable=no-value-for-parameter
+        self._current_qjob = backend._submit_job(
+            qobj=FakeQobj())  # pylint: disable=no-value-for-parameter
```

Comment:
How would I create a fake circuit? When using a real circuit `backend.run(circuit)` the test cases still fail - since a mock backend is used there are type errors "between instances of 'Mock' and 'int'" in the `assemble` method

Reply:
You'd need to figure out why it's failing and add the proper mock as needed.